content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,142 @@
+Wed Jun  5 19:11:15 PDT 2019 create new ca secret
+secret/istio-ca-secret created
+pod "istio-citadel-8574b88bcd-j7v2d" deleted
+Wed Jun  5 19:11:18 PDT 2019 restarted Citadel, checking status
+NAME                             READY     STATUS    RESTARTS   AGE
+istio-citadel-8574b88bcd-l2g74   1/1       Running   0          3s
+New root certificate:
+Certificate:
+    Data:
+        ...
+        Validity
+            Not Before: Jun  6 03:24:43 2019 GMT
+            Not After : Jun  3 03:24:43 2029 GMT
+        Subject: O = " cluster.abc"
+        ...
+{{< /text >}}
+
+After the transition, the Envoy sidecars may be hot-restarted. This may have some impact on your traffic.
+Please refer to 
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5) for more details.
+
+### Step 2. Verify the new workload certificates are generated
+
+Run the following command to verify that all the new certificates are generated. If not, wait a minute and run it again.
+
+{{< text bash>}}
+$ ./root_transition.sh verify
+This script checks the current root CA certificate is propagated to all the Istio-managed workload secrets in the cluster.                    
+Root cert MD5 is 8fa8229ab89122edba73706e49a55e4c
+Checking namespace: default
+  Secret default.istio.default is updated.
+  Secret default.istio.sleep is updated.
+Checking namespace: istio-system
+  Secret istio-system.istio.default is updated.
+  ...
+------All Istio keys and certificates are updated in secret!
+{{< /text >}}
+
+### Step 3. Install Istio 1.1.8
+
+Note that this step is necessary, to make sure the control plane components and sidecar Envoys all reload the new certificates and keys.
+
+Upgrade your control plane and Envoy sidecars to 1.1.8. You can refer to this
+[doc](https://istio.io/docs/setup/kubernetes/upgrade/steps/)
+to do the upgrades.
+
+### Step 4. Verify the new workload certificates are loaded by Envoy
+
+You can verify whether an Envoy has received the new certificates.
+The following command shows an example to check the Envoy’s certificate for pod _foo_ running in namespace _bar_.
+
+{{< text bash>}}
+$ kubectl exec -it foo -c istio-proxy -n bar -- curl http://localhost:15000/certs | head -c 1000
+{
+ "certificates": [
+  {
+   "ca_cert": [
+      ...
+      "valid_from": "2019-06-06T03:24:43Z",
+      "expiration_time": ...
+   ],
+   "cert_chain": [
+    {
+      ...
+    }
+{{< /text >}}
+
+Please inspect the "valid\_from" value of the "ca\_cert".
+If it matches the "_Not_ _Before_" value in the new certificate as shown in Step 1,
+your Envoy has loaded the new root certificate.
+
+## Troubleshooting
+
+### Why my workloads do not pick up the new certificates (in Step 4)?
+
+Please make sure you have updated to Istio 1.1.8 for the sidecars in Step 3.
+If you are using Istio releases 1.1.3 - 1.1.7, pilot agent may not hot-restart Envoy
```

Comment:
this demands a warning callout.

Reply:
pilot agent/the pilot agent

Reply:
The community generally think of `isito-proxy` rather than `pilot-agent` I think; might be good to clarify what you mean.

And there's no "may not" , it is "will not"

Reply:
@stone, I prefer making it accurate. How about adding a hyper link https://istio.io/docs/reference/commands/pilot-agent/ here?

Reply:
Removed the Pilot agent concept to reduce the confusion.
It's not 100% "will not", because for the first time the certificates change, the Pilot agent can restart Envoy. It doesn't do that after the first time.